python/plugins/processing/algs/gdal/translate.py

```diff
@@ -80,11 +80,11 @@ def defineCharacteristics(self):
                                           self.tr("Nodata value, leave blank to take the nodata value from input"),
                                           '', optional=True))
         self.addParameter(ParameterSelection(self.EXPAND,
-                                             self.tr('Expand'), ['none', 'gray', 'rgb', 'rgba']))
+                                             self.tr('Expand'), ['none', 'gray', 'rgb', 'rgba'], default=0))
         self.addParameter(ParameterCrs(self.SRS,
                                        self.tr('Output projection for output file [leave blank to use input projection]'), None, optional=True))
         self.addParameter(ParameterExtent(self.PROJWIN,
-                                          self.tr('Subset based on georeferenced coordinates')))
+                                          self.tr('Subset based on georeferenced coordinates'), optional=True))
```

Comment:
That's what I thought, but when I run 
`processing.runalg('gdalogr:translate', {'INPUT': 'c:\data\test.tif', 'EXPAND': 0})`
in QGIS 2.18.3, I get the following error message:
`Error: Missing parameter value for parameter PROJWIN.`

Adding `optional=True` fixes this. In the `warp` algorithm, the same change was made a couple of months ago in 3b1dfa7071f7ddd9b2164b0d795141612a93d302